nsq/async.py

```diff
@@ -241,7 +252,7 @@ def close(self):
     def _read_size(self, data):
         try:
             size = struct.unpack('>l', data)[0]
-            self.stream.read_bytes(size, self._read_body)
+            self._read_bytes(size, self._read_body)
         except Exception:
             self.close()
```

Comment:
Based on the original code it looked like the exception was to handle struct unpacking and that self.close() was there if the struct unpacking threw an exception. So now if the unpack fails the stream should not be closed? I don't know enough about the low level details, but it would be a different behavior than the original code.